scripts/tests/scripts_regression_tests.py

```diff
@@ -2506,36 +2511,45 @@ def _main_func():
     elif "CIME_MACHINE" in os.environ:
         mach_name = os.environ["CIME_MACHINE"]
         MACHINE = Machines(machine=mach_name)
+    elif config.has_option("create_test", "MACHINE"):
+        MACHINE = Machines(machine=config.get("create_test", "MACHINE"))
+    elif config.has_option("main", "MACHINE"):
+        MACHINE = Machines(machine=config.get("main", "MACHINE"))
     else:
         MACHINE = Machines()
 
-
     if "--compiler" in sys.argv:
-        global TEST_COMPILER
         midx = sys.argv.index("--compiler")
         TEST_COMPILER = sys.argv[midx + 1]
         del sys.argv[midx + 1]
         del sys.argv[midx]
+    elif config.has_option("create_test", "COMPILER"):
```

Comment:
No environment variable here? You promised me a `CIME_COMPILER` environment variable! Sniff :(

> I lied - there is no CIME_COMPILER environment variable. Working on this now.

`elif "CIME_COMPILER" in os.environ:`

Reply:
Why?  I don't think that we need one.  When I said working on this I meant I was addressing the issue of the PR not that I was creating an environment variable.

Reply:
Fair enough -- I just misunderstood (but who doesn't love more environment variables or 16 ways to do the same thing? :)